File: frontend/src/api/models/ChatCompletionRequestDTO.ts
import type { ChatMessageDTO } from './ChatMessageDTO';

export interface ChatCompletionRequestDTO {
  messages: ChatMessageDTO[];
}

export function ChatCompletionRequestDTOToJSON(
  value: ChatCompletionRequestDTO,
): any {
  return value;
}

export function ChatCompletionRequestDTOFromJSON(
  json: any,
): ChatCompletionRequestDTO {
  return json as ChatCompletionRequestDTO;
}

File: backend/src/main/java/ch/uzh/ifi/imrg/platform/rest/dto/input/ChatCompletionRequestDTO.java
package ch.uzh.ifi.imrg.platform.rest.dto.input;

import ch.uzh.ifi.imrg.platform.enums.Language;
import java.util.List;
import lombok.Data;

@Data
public class ChatCompletionRequestDTO {
  private List<ChatMessageDTO> messages;
  private Language language;
}
